mwdb/core/search/fields.py

```diff
@@ -26,348 +28,360 @@
 
 from .exceptions import (
     FieldNotQueryableException,
+    InvalidValueException,
     ObjectNotFoundException,
-    UnsupportedGrammarException,
+    UnsupportedLikeStatement,
+    UnsupportedNodeTypeException,
+)
+from .parse_helpers import (
+    PathSelector,
+    has_wildcard,
+    jsonpath_config_string_equals,
+    jsonpath_range_equals,
+    jsonpath_string_equals,
+    make_jsonpath_selector,
+    range_equals,
+    string_equals,
+    transform_for_config_like_statement,
+    transform_for_like_statement,
+    transform_for_quoted_config_like_statement,
+    transform_for_quoted_like_statement,
+    unescape_string,
 )
-from .tree import Subquery
-
-Expression = Union[Range, Term]
 
 
-def get_term_value(node: Term) -> str:
-    """
-    Retrieves unescaped value from Term with transformed wildcards
-    to the SQL form
-    :param node: luqum.Term object
-    :return: Unescaped value
-    """
-    if node.has_wildcard():
-        wildcard_map = {"*": "%", "?": "_"}
-
-        # Escape already contained SQL wildcards
-        node_value = re.sub(r"([%_])", r"\\\1", node.value)
-        # Transform unescaped Lucene wildcards to SQL form
-        node_value = Term.WILDCARDS_PATTERN.sub(
-            lambda m: wildcard_map[m.group(0)], node_value
-        )
-        # Unescape Lucene escaped special characters
-        node_value = Term.WORD_ESCAPED_CHARS.sub(r"\1", node_value)
-        return node_value
+def string_from_node(node: Item, escaped: bool = False) -> str:
+    if isinstance(node, Word):
+        return node.value if escaped else node.unescaped_value
+    elif isinstance(node, Phrase):
+        # Remove quotes from the beginning and the end of Phrase
+        return node.value[1:-1] if escaped else node.unescaped_value[1:-1]
     else:
-        return node.unescaped_value
+        raise UnsupportedNodeTypeException(node)
+
+
+def range_from_range_node(
+    node: Range,
+) -> Tuple[Optional[str], Optional[str], bool, bool]:
+    low_value = string_from_node(node.low)
+    if low_value == "*":
+        low_value = None
+    elif has_wildcard(low_value):
+        raise UnsupportedLikeStatement(node.low)
```

Comment:
I think I understand this, but it's still a bit magic. Maybe add a docstring to this function with example of what's parsed? Adding a better type for the return value may also help.

Reply:
I renamed `has_wildcard` to `is_pattern_value` and `UnsupportedPatternValue`. In simple words: we don't accept values with wildcards unless it's single wildcard.